### PR TITLE
ucx/pml: show warning if already unsupported UCX version is used

### PR DESCRIPTION
**Pull request by karasevb**

### CODE REVIEW

File: config/ompi_check_ucx.m4 [hunk 1]
```diff
@@ -74,8 +74,15 @@ AC_DEFUN([OMPI_CHECK_UCX],[
                              [ompi_check_ucx_cv_have_version_1_8=yes])])
            AS_IF([test "${ompi_check_ucx_cv_have_version_1_8}" = "yes"],
                  [AC_MSG_WARN([UCX support skipped because version 1.8.x was found, which has a known catastrophic issue.])
-                  AC_MSG_WARN([Please upgrade to UCX version 1.9 or higher.])
                   ompi_check_ucx_happy=no])])
+           AC_PREPROC_IFELSE([AC_LANG_PROGRAM([[
+#include <ucp/api/ucp_version.h>
+                             ]], [[
+#if (UCP_API_MAJOR <= 1) && (UCP_API_MINOR < 9)
```
Comment (hoopoepg) [inline config/ompi_check_ucx.m4:81]: will it work on UCX version 0.10?

File: config/ompi_check_ucx.m4 [hunk 2]
```diff
@@ -74,8 +74,15 @@ AC_DEFUN([OMPI_CHECK_UCX],[
...
+#error "Version too low"
+#endif
+                             ]])],
+                             [], [AC_MSG_WARN([UCX version is too old, please upgrade to 1.9 or higher.])])
```
Comment (brminich) [inline config/ompi_check_ucx.m4:85]: why need to change this?

File: config/ompi_check_ucx.m4 (hunk 1 above)
Comment (karasevb) [reply config/ompi_check_ucx.m4]: considering that version 0.10 doesn't exist - no, fixed to compare starting strictly from 1

File: config/ompi_check_ucx.m4 (hunk 2 above)
Comment (karasevb) [reply config/ompi_check_ucx.m4]: would be helpful to see the warning message during `./configure` time, and not just runtime

File: config/ompi_check_ucx.m4 [hunk 3]
```diff
@@ -74,8 +74,15 @@ AC_DEFUN([OMPI_CHECK_UCX],[
                              [ompi_check_ucx_cv_have_version_1_8=yes])])
            AS_IF([test "${ompi_check_ucx_cv_have_version_1_8}" = "yes"],
                  [AC_MSG_WARN([UCX support skipped because version 1.8.x was found, which has a known catastrophic issue.])
-                  AC_MSG_WARN([Please upgrade to UCX version 1.9 or higher.])
                   ompi_check_ucx_happy=no])])
+           AC_PREPROC_IFELSE([AC_LANG_PROGRAM([[
+#include <ucp/api/ucp_version.h>
+                             ]], [[
+#if (UCP_API_MAJOR == 1) && (UCP_API_MINOR < 9)
```
Comment (hoopoepg) [inline config/ompi_check_ucx.m4:81]:
```suggestion
#if (UCP_API_MAJOR < 1) || ((UCP_API_MAJOR == 1) && (UCP_API_MINOR < 9))
```

**Review by hoopoepg — APPROVED** [review]

**Review by brminich — APPROVED** [review]

**Comment by awlauria** [comment]
please bring back to v5.0.x

**Pull request merged**